src/mlpack/tests/layer_test.cpp

```diff
@@ -0,0 +1,277 @@
+/**
+ * @file inception_layer_test.cpp
+ * @author Nilay Jain
+ *
+ * Tests the inception layer.
+ */
+#include <mlpack/core.hpp>
+
+#include <mlpack/methods/ann/activation_functions/logistic_function.hpp>
+
+#include <mlpack/methods/ann/layer/one_hot_layer.hpp>
+#include <mlpack/methods/ann/layer/conv_layer.hpp>
+#include <mlpack/methods/ann/layer/pooling_layer.hpp>
+#include <mlpack/methods/ann/layer/softmax_layer.hpp>
+#include <mlpack/methods/ann/layer/bias_layer.hpp>
+#include <mlpack/methods/ann/layer/linear_layer.hpp>
+#include <mlpack/methods/ann/layer/base_layer.hpp>
+#include <mlpack/methods/ann/layer/inception_layer.hpp>
+#include <mlpack/methods/ann/layer/concat_layer.hpp>
+#include <mlpack/methods/ann/performance_functions/mse_function.hpp>
+#include <mlpack/core/optimizers/rmsprop/rmsprop.hpp>
+
+#include <mlpack/methods/ann/init_rules/random_init.hpp>
+#include <mlpack/methods/ann/cnn.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using namespace mlpack;
+using namespace mlpack::ann;
+using namespace mlpack::optimization;
+
+
+BOOST_AUTO_TEST_SUITE(InceptionLayerTest);
+
+void Test(arma::mat m1, arma::mat m2)
+{
+  for (size_t i = 0; i < m1.n_cols; ++i)
```

Comment:
>I am comparing all columns of m1 with all columns of m2. Is this incorrect?

Nope, because m1(i) access single data of the arma::mat but not "a column of the mat".
If you want to make sure every columns have the same value, I think it is same as make sure all of the data as equal, change it to n_elem should do the tricks.